refnx/reduce/platypusnexus.py

```diff
@@ -498,6 +498,142 @@ def _chopper_values(self, h5data):
         return master, slave, speeds[0] / 60.0, phases[slave - 1]
 
 
+class PolarisedCatalogue(PlatypusCatalogue):
+    """
+    Extract relevant parts of a polarised PLATYPUS NeXus file
+    for reflectometry reduction
+    """
+
+    def __init__(self, h5d):
+        """
+        Extract relevant parts of a polarised PLATYPUS NeXus file
+        for reflectometry reduction
+        Access information via dict access, e.g. cat['pol_flip_freq'].
+
+        Parameters
+        ----------
+        h5d - HDF5 file handle
+        """
+        super(PolarisedCatalogue, self).__init__(h5d)
+        # Is there a magnet?
+        self.ismagnet = False
+        # Is there a cryocooler?
+        self.iscryo = False
+        # Is there a power supply?
+        self.isPS = False
+
+        d = self.cat
+
+        d = self._polarised_flippers(d, h5d)
+        d = self._analyser_flippers(d, h5d)
+        d = self._analyser_params(d, h5d)
+        d = self._check_sample_environments(d, h5d)
+
+    def _polarised_flippers(self, d, h5d):
+        d["pol_flip_freq"] = h5d[
+            "entry1/instrument/polarizer_flipper/flip_frequency"
+        ][0]
+        d["pol_flip_current"] = h5d[
+            "entry1/instrument/polarizer_flipper/flip_current"
+        ][0]
+        d["pol_flip_voltage"] = h5d[
+            "entry1/instrument/polarizer_flipper/flip_voltage"
+        ][0]
+        d["pol_flip_status"] = h5d[
+            "entry1/instrument/polarizer_flipper/flip_on"
+        ][0]
+        d["pol_guide_current"] = h5d[
+            "entry1/instrument/polarizer_flipper/guide_current"
+        ][0]
+        return d
+
+    def _analyser_flippers(self, d, h5d):
+        d["anal_flip_freq"] = h5d[
+            "entry1/instrument/analyzer_flipper/flip_frequency"
+        ][0]
+        d["anal_flip_current"] = h5d[
+            "entry1/instrument/analyzer_flipper/flip_current"
+        ][0]
+        d["anal_flip_voltage"] = h5d[
+            "entry1/instrument/analyzer_flipper/flip_voltage"
+        ][0]
+        d["anal_flip_status"] = h5d[
+            "entry1/instrument/analyzer_flipper/flip_on"
+        ][0]
+        d["anal_guide_current"] = h5d[
+            "entry1/instrument/analyzer_flipper/guide_current"
+        ][0]
+        return d
+
+    def _analyser_params(self, d, h5d):
+        d["analyzer_base"] = h5d["entry1/instrument/polarizer/anal_base"][0]
```

Comment:
Are these really needed to be loaded for data reduction purposes? They're important to know for acquisition, but I can't see the value otherwise.

Reply:
No, I don't think any of these are needed very often at all - this whole function can be deleted.